### ISSUE TITLE
[BUG] S3 Cloud recording

### ISSUE THREAD

**Issue by BrainRift** [AgoraIO/Agora-WordPress#90]
After ensuring the proper S3 credentials and bucket write, I am unable to record a session.  I receive an error, but after turning debugging in Wordpress, the error is never displayed or logged....
<img width="431" alt="Screen Shot 2020-11-30 at 9 07 04 PM" src="https://user-images.githubusercontent.com/10853812/100691002-bb260580-3355-11eb-808b-49425c430b4b.png">

**Desktop (please complete the following information):**
 - OS: Mac 11.0.1
 - Browser: Chrome 
 - Version: 86.0.4240.198 (Official Build) (x86_64)

I am running the webserver via AWS Lightsail on a bitnami Wordpress instance
Everything else appears to work just fine.

**Comment by digitallysavvy** [comment]
@BrainRift Cloud recording could be failing for a number of reasons. To ensure you are properly set up on the Agora side, did you set up cloud recording on your Agora Project ([as outlined in this guide](https://medium.com/agora-io/agora-cloud-recording-quickstart-guide-with-postman-demo-c4a6b824e708))?

**Comment by digitallysavvy** [comment]
@BrainRift just checking to see if the guide helped resolve your issue or are you still facing this issue?

**Comment by ray45874587** [comment]
I also encountered this problem. I set up aliyun OSS, and other functions were normal. Only when I clicked to record, the same error message as BrainRift appeared @digitallysavvy

**Comment by ray45874587** [comment]
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function WP_Agora_Public::generateNewToken(), 2 passed in /www/wwwroot/wp-content/plugins/wp-agora-io/includes/class-wp-agora-cloud-recording.php on line 186 and exactly 3 expected in /www/wwwroot/wp-content/plugins/wp-agora-io/public/class-wp-agora-io-public.php:122
Stack trace:
#0 /www/wwwroot/wp-content/plugins/wp-agora-io/includes/class-wp-agora-cloud-recording.php(186): WP_Agora_Public->generateNewToken()
#1 /www/wwwroot/wp-content/plugins/wp-agora-io/includes/class-wp-agora-cloud-recording.php(43): AgoraCloudRecording->startRecording()
#2 /www/wwwroot/wp-includes/class-wp-hook.php(287): AgoraCloudRecording->ajaxAgoraCloudRecording()
#3 /www/wwwroot/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters()
#4 /www/wwwroot/wp-includes/plugin.php(484): WP_Hook->do_action()
#5 /www/wwwroot/wp-admin/admin-ajax.php(184): do_action()
#6 {ma in /www/wwwroot/wp-content/plugins/wp-agora-io/public/class-wp-agora-io-public.php on line 122

**Comment by digitallysavvy** [comment]
What value are you passing for the bucket name? If you are using S3, the ARN is listed as `arn:aws:s3:::bucketname` but with Agora's Cloud Recording you only need to use the `bucketname` and can omit `arn:aws:s3:::`

**Comment by ottverse** [comment]
Hi @digitallysavvy - I am facing the same issue and I get the same error "There has been a critical error on your website".

I followed the guide you provided in an earlier reply and enabled recording for the Agora project. I created an S3 bucket, blocked all public access, created an IAM user with all S3 access rights, and used that user's access + secret key for the recording. The S3 bucket is in US East Coast 2. For the Bucket name, I am not providing `arn:was:s3:::` as instructed - just the name of the bucket. 

As a quick test, I set up a new, dummy installation of WordPress and the same error occurs there as well. I am using DigitalOcean btw and all other functionalities with WordPress and Agora work fine.
